### PR TITLE
Allow attributes to specify how to be represented in the frame repr

### PR DESCRIPTION
**Pull request by Cadair**
This implements a hook to customise the way an attribute is represented in the frame repr. The motivation for this is in https://github.com/sunpy/sunpy/pull/2723. 

Which to summarise is basically we want to be able to represent a coordinate frame as the celestial body it was generated from rather than the full coordinate frame that it is, but only really in the SkyCoord / Frame repr, the rest of the time we would quite like it to be a normal frame.

Basically that

```
'earth'
```

is much more meaningful to most humans than:

```
<SkyCoord (HeliographicStonyhurst: obstime=2018-08-17 17:13:20.523028): (lon, lat, radius) in (deg, deg, AU)
    (0., 6.75658585, 1.01238948)>
```

This is just to test the water, I am happy to actually finish this PR if there are no major objections!

### CODE REVIEW

**Comment by pllim** [comment]
Nitpick: `'Earth'` (capitalized) 😉

**Comment by Cadair** [comment]
I decided that the name should probably have `astropy` in the name to make it a little more obvious what it is doing (given you might want to implement this on any object anywhere).

Where would you suggest I put the docs for this? I couldn't really find a natural home.

File: CHANGES.rst [hunk 1]
```diff
@@ -37,6 +37,10 @@ astropy.coordinates
 
 - Added a ``concatenate_representations`` function to combine coordinate
   representation data and any associated differentials. [#7922]
+- ``BaseCoordinateFrame`` will now check for a method named
```
Comment (eteq) [inline CHANGES.rst]: (minor:) missing a space above this bullet

**Comment by Cadair** [comment]
I am not really sure how it is linked to representations @eteq It's only for attributes (not sure you have representations as attributes anywhere right?).

Anyway, have a look at what I have written.

**Comment by eteq** [comment]
@Cadair - I was going to leave a comment but decided it was easier to just write it myself.  If you are happy with that I think it's good to merge!

**Comment by Cadair** [comment]
Lgtm

**Comment by Cadair** [comment]
Thanks!
